apis/projectcontour/v1/httpproxy.go

```diff
@@ -353,6 +353,43 @@ type Route struct {
 	// The policy for rate limiting on the route.
 	// +optional
 	RateLimitPolicy *RateLimitPolicy `json:"rateLimitPolicy,omitempty"`
+
+	// RequestRedirectPolicy defines an HTTP redirection.
+	// +optional
+	RequestRedirectPolicy *HTTPRequestRedirectPolicy `json:"requestRedirectPolicy,omitempty"`
```

Comment:
The only issue I see with the API is that `services` is still required to have at least one entry, even if you just want a redirect. We could make `services` optional, and then check in the DAG build that either `services` or `requestRedirectPolicy` is specified. That does have the downside of slower feedback to the user though (at DAG build time rather than at HTTPProxy creation time). I _think_ I'd probably still make that change, as having to put dummy data into `services` when you want a redirect feels clumsy, but interested in what others think.

Reply:
Agreed, having to specify a service when the traffic won't get there feels a bit weird.

Making a required field optional is a compatible API change (because old objects will continue to work), so we can conceivably make `services` optional without an API version increment. So I guess I'm saying I'm a +1 to making `services` optional.

Reply:
That's a good point @skriss! Looks like it's just a kubebuilder default that we could change to enable not requiring services to be present.